src/sentry/incidents/grouptype.py

```diff
@@ -27,29 +35,136 @@ class MetricIssueEvidenceData(EvidenceData):
     alert_id: int
 
 
-class MetricAlertDetectorHandler(StatefulDetectorHandler[QuerySubscriptionUpdate, int]):
+class MetricIssueDetectorHandler(StatefulDetectorHandler[QuerySubscriptionUpdate, int]):
     def create_occurrence(
         self,
         evaluation_result: ProcessedDataConditionGroup,
         data_packet: DataPacket[QuerySubscriptionUpdate],
         priority: DetectorPriorityLevel,
-    ) -> tuple[DetectorOccurrence, dict[str, Any]]:
-        # Returning a placeholder for now, this may require us passing more info
-        occurrence = DetectorOccurrence(
-            issue_title="Some Issue Title",
-            subtitle="An Issue Subtitle",
-            type=MetricIssue,
-            level="error",
-            culprit="Some culprit",
+    ) -> tuple[DetectorOccurrence, EventData]:
+        try:
+            alert_rule_detector = AlertRuleDetector.objects.get(detector=self.detector)
+            alert_id = alert_rule_detector.alert_rule_id
+        except AlertRuleDetector.DoesNotExist:
+            alert_id = None
+
+        try:
+            detector_trigger = DataCondition.objects.get(
+                condition_group=self.detector.workflow_condition_group, condition_result=priority
+            )
+        except DataCondition.DoesNotExist:
+            raise DetectorException(
+                f"Failed to find detector trigger for detector id {self.detector.id}, cannot create metric issue occurrence"
+            )
+
+        try:
+            query_subscription = QuerySubscription.objects.get(id=data_packet.source_id)
+        except QuerySubscription.DoesNotExist:
+            raise DetectorException(
+                f"Failed to find query subscription for detector id {self.detector.id}, cannot create metric issue occurrence"
+            )
+
+        try:
+            snuba_query = SnubaQuery.objects.get(id=query_subscription.snuba_query_id)
+        except SnubaQuery.DoesNotExist:
+            raise DetectorException(
+                f"Failed to find snuba query for detector id {self.detector.id}, cannot create metric issue occurrence"
+            )
+
+        try:
+            assignee = parse_and_validate_actor(
+                str(self.detector.created_by_id), self.detector.project.organization_id
+            )
+        except Exception:
+            assignee = None
+
+        title = self.construct_title(snuba_query, detector_trigger, priority)
+        event_data = {
+            "project_id": self.detector.project_id,
+            "environment": self.detector.config.get("environment"),
+            "platform": None,
+            "sdk": None,
+        }  # XXX: may need to add to this
```

Comment:
some of these are done automatically in the stateful detector itself. (https://github.com/getsentry/sentry/blob/master/src/sentry/workflow_engine/handlers/detector/stateful.py#L418-L423)

Is there anything that we could do to illustrate what's provided / what we'd want? (I was thinking of improving the types around `EventData` but i'm not 100% sure what all is expected to be in this object for the Issue Platform)

Reply:
Hmm I think it was just tough cause there were no examples so I was looking at how we make issue occurrences in random other parts of the codebase.

Reply:
Could we have `issue_title=self.detector.name` set on the stateful base class?